QuestionBot/QuestionBot/Model/InMemoryStore.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace QuestionBot.Model {
+
+    public class InMemoryStore : IStore {
+        private IList<IRecord> allRecords = new List<IRecord>();
+        private static int _questionId = 0;
+
+        public IRecord CreateRecord(string question)
+        {
+            _questionId++;
+
+            Record newQuestion = new Record(_questionId, question, DateTime.Now);
+            allRecords.Add(newQuestion);
+
+            return newQuestion;
+        }
+
+        public IRecord UpdateRecord(int id, string answer)
+        {
+            IRecord recordToUpdate = allRecords.FirstOrDefault(recordItem => recordItem.ID == id);
```

Comment:
I think I was using ``` FirstOrDefault ``` here so that if the ID doesn't exist, it doesn't throw an exception, it will return the default value of null... But I think I should really just put a try catch here and use the exception throwing to my advantage using ``` .First() ``` . 
Hard to remember what my thought process was but I think that was it :stuck_out_tongue: